partner_tlr_update/models/res_partner.py

```diff
@@ -213,25 +215,39 @@ def update_contact_person(self, xml, parent_id):
                     'login': 'FromTLR'
                 }])
             user.partner_id.update_from_xml(xml, 'contact_persons')
-            employee = self.env['hr.employee'].create({
-                'name': user.name,
-                'address_id': parent_id,
-                })
-            email = user.partner_id.email
-            if not email:
-                email = ""
-            if self.env['res.users'].search_count([('login', '=', email)]):
-                partner = user.partner_id
-                user.unlink()
-                partner.unlink()
-                employee.unlink()
-                return
-            user.write({
-                'employee_ids': [(6, 0, [employee.id])],
-                'login': "_".join((
-                    email,
-                    user.partner_id.legacy_no))
+            if not children:
+                employee = self.env['hr.employee'].create({
+                    'name': user.name,
+                    'address_id': parent_id,
+                    })
+                email = user.partner_id.email
+                login = email
+                if email:
+                    # in case there is already another user with the same email for some reason
```

Comment:
Row to long, as it is a comment it's easy to split on several rows.
Comment rows have the limit of 72 chars